sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -117,6 +121,16 @@ private static void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 messageArgs: MemberNameToMessageArguments[memberAccessNameName]));
         }
 
+        private static bool IsResultInContinueWithCall(string memberAccessName, MemberAccessExpressionSyntax memberAccess) =>
+            memberAccessName == ResultName &&
+            memberAccess.Expression is IdentifierNameSyntax identifierNameSyntax &&
+            memberAccess.FirstAncestorOrSelf<InvocationExpressionSyntax>(invocation => IsContinueWithCallWithArgumentName(invocation, identifierNameSyntax.GetName())) != null;
```

Comment:
I suggest to extract result of `GetName()` to variable. Invocation chain like `A().B().C().D()` could create long ancestor chain to iterate.

Reply:
AFAICT GetName() is called on identifierNameSyntax and we don't have a long chain.

As it used in only one place I prefer to keep this as it is. It also keeps the code simpler.

Reply:
Clarification of my previous statement from offline discussion:
If someone does `A(B(C(D(task.Result))))` then then `GetName` would be called in a loop instead of once.